accepted/2024/supplied-by-platform.md

```diff
@@ -0,0 +1,217 @@
+The feature may be available in minor versions of the .NET 9 SDK as disabled with the above opt-in available.
+
+### Technical explanation
+
+A list of package id/versions to be pruned will be provided by the .NET SDK.
+To aid this, a new item type and collect targets will be introduced.
+
+The `PrunedPackageReference` item will support the following attributes:
+
+| Attribute | Explanation |
+|-----------|-------------|
+| Version | A NuGet parsable version. The version is consider to the maximum version to be pruned. |
+
+The collect target will be `CollectPrunedPackageReferences`.
+
+When NuGet sees any of these package ids in the resolution, it'll just skip them and log a detailed verbosity message, indicating the package has been skipped.
+Pruning direct PackageReference is *not allowed*.
+We may capture some of this information in telemetry to track how often this feature is being used.
+
+#### Changes to the obj files
+
+`project.assets.json`
+
+```json
+{
+    "NuGet.LibraryModel/6.10.0": {
+        "type": "package",
+        "dependencies": {
+          "NuGet.Common": "6.10.0",
+          "NuGet.Versioning": "6.10.0"
+        },
+        "compile": {
+          "lib/netstandard2.0/NuGet.LibraryModel.dll": {}
+        },
+        "runtime": {
+          "lib/netstandard2.0/NuGet.LibraryModel.dll": {}
+        }
+      },
+      "NuGet.Packaging/6.10.0": {
+        "type": "package",
+        "dependencies": {
+          "Newtonsoft.Json": "13.0.3",
+          "NuGet.Configuration": "6.10.0",
+          "NuGet.Versioning": "6.10.0",
+          "System.Security.Cryptography.Pkcs": "6.0.4"
+        },
+        "compile": {
+          "lib/net5.0/NuGet.Packaging.dll": {}
+        },
+        "runtime": {
+          "lib/net5.0/NuGet.Packaging.dll": {}
+        }
+      }
+}
+```
+
+The above represents the targets section for a framework.
+NuGet.Versioning is pruned. NuGet.Versioning will appear as a dependency, but it will not be in the list since it was never chosen.
+
+`PackageSpec & project section of the assets file`
+
+TODO NK
+
+### Additional validation considerations
+
+- How does leaving the dependency in the assets file section affect features consuming the assets file.
+  - The solution explorer tree - Prototype shows the solution explorer is skipping the reference.
+  - PM UI tab - Prototype shows that since the reference does not really exist, it is not show at all.
+  - list package - Prototype shows that since the reference does not really exist, it is not show at all.
+  - dotnet nuget why - TODO NK
+
+### NET SDK - selecting packages to be pruned
+
+See [Prior art](#net-sdk-package-pruning) for more details on how the .NET SDK prunes packages at runtime today.
+The particular pruning is based on the resulting list of shared frameworks.
+The .NET SDK will provide the list at the beginning of the restore operation, as such the .NET SDK *must* only consider direct shared frameworks, and not transitive ones.
```

Comment:
Honestly I've only ever seen folks complain when they get transitive shared frameworks.  It's often accidental and it breaks deployment -- @dsplaisted what if the SDK encouraged people to make them direct?

IOW if the result of a restore or resolving project references changed the shared framework of the project the SDK would warn the user and ask them explicitly reference that framework.

Reply:
I'm wary of changing that behavior, because we don't know how many people we aren't hearing from who are relying on it (knowingly or not).

With respect to the supplied by platform design, I think it would probably be OK to only consider the direct framework references.

Reply:
What if we made it just a warning:
1. When transitive framework was brought in after restore.
2. We can see that the project has packages that overlap with that framework.
3. Warn that those packages could be dropped by directly referencing the framework.
4.

Reply:
I like that idea. 
I added it in future possibilities.